### PR TITLE
Asset.get_path frontend changes

### LINKED ISSUE #600: DJANGO_API: takes awhile to get to the 2nd level subdirectory for a dandiset
To facilitate future support of large (in # of files, https://github.com/dandi/dandi-api/issues/78) dandisets I am in the midst of uploading a test dandiset with 1,000,000 assets . By now (running for probably 18 hours) uploaded only about 43k according to the summary of https://gui-beta-dandiarchive-org.netlify.app/#/dandiset/010000/draft .

An interesting behavior already showed up.
- navigate: "View Data", "sub-1/" , "sub-1_ses-10_cell-1.z" -- *everything (besides minor UI glitch https://github.com/dandi/dandiarchive/issues/599) is working nicely and not too slow*
- now if you click on "sub-1" in the breadcrumbs navigator then
  - it would take you many (I waited for 20) seconds: *should be at least as fast as you went there originally (so not sure what holds it up)*
  - meanwhile breadcrumbs path would show the target destination (`sub-1`) while the files listing pane would show old sub-directory content (while "spinning bar" is still spinning and some times getting stuck even) - make view a big confusing: IMHO *should clear the pane* (before repopulating with the new listing)

### PR DESCRIPTION
**Pull request by mvandenburgh**
Updates the file browser component to make use of the additional data returned from the `Asset.get_path` endpoint introduced in https://github.com/dandi/dandi-api/pull/312.

These changes eliminate the slowdown that occurs in the file browser mentioned in #600 (closes #600).

Blocked until https://github.com/dandi/dandi-api/pull/312 is merged, so I'll mark this as draft for now.

### CODE REVIEW

File: web/src/views/FileBrowserView/PublishFileBrowser.vue [hunk 1]
```diff
@@ -188,7 +187,28 @@ export default {
...
+            });
+          });
+        }
+
+        if (data.folders) {
+          mapped = mapped.concat(Object.keys(data.folders).map(
+            (name) => ({ name: `${name}/`, folder: true, ...data.folders[name] }),
+          ));
+        }
+
```
Comment (AlmightyYakob) [inline web/src/views/FileBrowserView/PublishFileBrowser.vue:211]: Based on the above comments, we might be able to simplify this logic like so:

```typescript
        return [
          ...location !== rootDirectory ? [{ name: parentDirectory, folder: true }] : [],
          ...Object.keys(data.files).map(
            (key) => ({ ...data.files[key], name: key, folder: false }),
          ),
          ...Object.keys(data.folders).map(
            (key) => ({ ...data.folders[key], name: `${key}/`, folder: true }),
          ),
        ];
```

Comment (AlmightyYakob) [reply web/src/views/FileBrowserView/PublishFileBrowser.vue:211]: And thinking about this now, we should probably put folders first, so they're visible

**Review by AlmightyYakob — CHANGES_REQUESTED** [review]
Looks good, I tested it and it worked without issue. I recommended some changes that I just think would make the file browser code (most of it not yours) a bit cleaner.

File: web/src/views/FileBrowserView/PublishFileBrowser.vue [hunk 2]
```diff
@@ -188,7 +187,28 @@ export default {
...
+        if (data.files) {
+          mapped = Object.keys(data.files).map(
+            (name) => ({ name, folder: false, ...data.files[name] }),
+          );
+          // set download links and uuids (needed for deleting) for files
+          mapped.forEach((item) => {
+            this.$set(this.itemData, item.name, {
+              download: item.download_url,
+              uuid: item.asset_id,
+            });
```
[the remark this answers is not in the document]
Comment (mvandenburgh) [reply web/src/views/FileBrowserView/PublishFileBrowser.vue]: I refactored things to remove `itemData`, and I was also able to remove `dialogName`. https://github.com/dandi/dandiarchive/pull/773/commits/6d95496787f71bb457d1586365dab6b046044aa0

File: web/src/views/FileBrowserView/PublishFileBrowser.vue (hunk 1 above)
Comment (mvandenburgh) [reply web/src/views/FileBrowserView/PublishFileBrowser.vue]: https://github.com/dandi/dandiarchive/pull/773/commits/17b25cc61215faaf90c56b9a73e4c1d1904ec4d7

File: web/src/views/FileBrowserView/PublishFileBrowser.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (mvandenburgh) [reply web/src/views/FileBrowserView/PublishFileBrowser.vue]: Oops, just realized `dialogActive` can be removed too. https://github.com/dandi/dandiarchive/pull/773/commits/c051ab58ad3bc8f2b298b34cac27c3d19fd34f89

**Review by AlmightyYakob — APPROVED** [review]
LGTM, tested locally with no issue

**Comment by yarikoptic** [comment]
Given the fix https://github.com/dandi/dandi-api/pull/387/files#diff-2f7f82b6ec9c0331ea6b7f270afb58015baf10237df23dbf41fc5597f5001f75R27 , shouldn't this PR base against staging?

File: web/src/rest/publish.ts [hunk 3]
```diff
@@ -218,9 +218,8 @@ const publishRest = new Vue({
       const { data } = await client.get('stats/');
       return data;
     },
-    assetDownloadURI(identifier: string, version: string, asset: Asset) {
-      const { asset_id } = asset;
-      return `${publishApiRoot}dandisets/${identifier}/versions/${version}/assets/${asset_id}/download/`;
+    assetDownloadURI(identifier: string, version: string, uuid: string) {
+      return `${publishApiRoot}dandisets/${identifier}/versions/${version}/assets/${uuid}/download/`;
```
Comment (satra) [inline web/src/rest/publish.ts:222]:
```suggestion
      return `${publishApiRoot}assets/${uuid}/download/`;
```

**Pull request merged**